src/python_bindings/ar/bind_ar.cpp

```diff
@@ -13,16 +13,34 @@ namespace py = pybind11;
 
 namespace python_bindings {
 void BindAr(py::module_& main_module) {
+    using algos::Algorithm;
     using namespace algos;
+    using model::ArIDs;
     using model::ARStrings;
 
     auto ar_module = main_module.def_submodule("ar");
-    py::class_<ARStrings>(ar_module, "AssociativeRule")
+    py::class_<ARStrings>(ar_module, "ARStrings")
             .def("__str__", &ARStrings::ToString)
             .def_readonly("left", &ARStrings::left)
             .def_readonly("right", &ARStrings::right)
             .def_readonly("confidence", &ARStrings::confidence);
-    BindPrimitive<Apriori>(ar_module, &ARAlgorithm::GetArStringsList, "ArAlgorithm", "get_ars",
-                           {"Apriori"}, py::return_value_policy::move);
+
+    py::class_<ArIDs>(ar_module, "ArIDs")
+            .def_readonly("left", &ArIDs::left)
+            .def_readonly("right", &ArIDs::right)
+            .def_readonly("confidence", &ArIDs::confidence);
+
+    py::class_<ARAlgorithm, Algorithm>(ar_module, "ArAlgorithm")
+            .def("get_ars", &ARAlgorithm::GetArStringsList)
```

Comment:
Is it just me or was there a slightly different comment here earlier? I've been trying to make keep_alive, not reference_internal for 2-3 hours. I came to an impasse and decided to write about it, but here is another comment. I'm confused.

Reply:
Whoops, sorry about that, I did indeed change it. I'm actually not sure how things work here myself, since the pybind11 docs don't say how STL conversions interact with return value policies.

Reply:
Update: it looks like pybind11 applies the return value policy for every element of the list it creates in the STL conversion, which is exactly what is needed here. Just use `reference_internal`

Reply:
`keep_alive<0, 1>`. 0 means the return value, 1 is `this`. And the return value policy that's needed is actually `move` (or `copy`, but `move` is faster), as the list of dependencies may change on a second run of `execute`. `reference_internal` is incorrect.

Reply:
Yeah, I was trying add py::keep_alive<0, 1> hole day yesterday. And problem was that pybind11 create weak references of it, so list can`t be created with this objects. I know why this happening, but I don`t know why that happening  in this case.

Reply:
Looks complicated. If you aren't able find a solution, we can merge it with a (broken) `reference_internal` policy, then move all members like this to `std::shared_ptr` so that we can copy them properly.

Reply:
Sure, but I want to try to find a solution a little more in time. А little sporting interest. If I can't, I'll inform about it on Fridays conference call or earlier.